core/src/subgraph/registrar.rs

```diff
@@ -24,6 +26,7 @@ pub struct SubgraphRegistrar<P, S, SM> {
     node_id: NodeId,
     version_switching_mode: SubgraphVersionSwitchingMode,
     assignment_event_stream_cancel_guard: CancelGuard, // cancels on drop
+    perf_config: Arc<SubgraphPerfRules>,
```

Comment:
Does this need to be an `Arc`? It seems to me the registrar could just take ownership of the rules